plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialCount.java

```diff
@@ -4,63 +4,98 @@
 import com.denizenscript.denizen.nms.NMSVersion;
 import com.denizenscript.denizen.objects.MaterialTag;
 import com.denizenscript.denizencore.objects.Mechanism;
-import com.denizenscript.denizencore.objects.ObjectTag;
 import com.denizenscript.denizencore.objects.core.ElementTag;
-import com.denizenscript.denizencore.objects.properties.Property;
 import com.denizenscript.denizencore.objects.properties.PropertyParser;
 import org.bukkit.block.data.BlockData;
 import org.bukkit.block.data.type.*;
 
-public class MaterialCount implements Property {
+public class MaterialCount extends MaterialProperty<ElementTag> {
+    // TODO: once 1.21 is the minimum supported version, remove PinkPetals interface in favor of FlowerBed
 
-    public static boolean describes(ObjectTag material) {
-        if (!(material instanceof MaterialTag mat)) {
-            return false;
-        }
-        if (!mat.hasModernData()) {
-            return false;
-        }
-        BlockData data = mat.getModernData();
+    // <--[property]
+    // @object MaterialTag
+    // @name count
+    // @input ElementTag(Number)
+    // @description
+    // Controls the amount of pickles in a Sea Pickle material, eggs in a Turtle Egg material, charges in a Respawn Anchor material, candles in a Candle material, flowers in a flower bed, or leaves on the ground.

```

Comment:
`leaves on the ground` is a bit vague, specify that this is about leaf litter like the other options do.